src/hevm/src/EVM/SymExec.hs

```diff
@@ -0,0 +1,287 @@
+    , vmoptCaller = addr
+    , vmoptOrigin = ethrunAddress --todo: generalize
+    , vmoptCoinbase = 0
+    , vmoptNumber = 0
+    , vmoptTimestamp = 0
+    , vmoptBlockGaslimit = 0
+    , vmoptGasprice = 0
+    , vmoptDifficulty = 0
+    , vmoptGas = 0xffffffffffffffff
+    , vmoptGaslimit = 0xffffffffffffffff
+    , vmoptMaxCodeSize = 0xffffffff
+    , vmoptSchedule = FeeSchedule.istanbul
+    , vmoptCreate = False
+    }) & set (env . contracts . at (createAddress ethrunAddress 1))
+             (Just (contractWithStore x initStore))
+
+-- Interpreter which explores all paths at
+-- branching points.
+-- returns a list of possible final evm states
+interpret
+  :: (EVM.Query -> IO (EVM ()))
+  -> Maybe Integer --max iterations
+  -> Stepper a
+  -> StateT VM IO (Either Stepper.Failure [a])
+interpret fetcher maxIter =
+  eval . Operational.view
+
+  where
+    eval
+      :: Operational.ProgramView Stepper.Action a
+      -> StateT VM IO (Either Stepper.Failure [a])
+
+    eval (Operational.Return x) =
+      pure (Right [x])
+
+    eval (action Operational.:>>= k) =
+      case action of
+        Stepper.Exec ->
+          exec >>= interpret fetcher maxIter . k
+        Stepper.Run ->
+          run >>= interpret fetcher maxIter . k
+        Stepper.Option (EVM.PleaseChoosePath continue) ->
+          do vm <- get
+             case maxIter of
+               Just maxiter ->
+                 let pc' = view (state . pc) vm
+                     addr = view (state . contract) vm
+                     iters = view (iterations . at (addr, pc') . non 0) vm
+                 in if num maxiter <= iters then
+                      vm ^?! (cache . path . ix ((addr, pc'), iters - 1)) & \case
+                        -- When we have reached maxIterations, we take the choice that will hopefully
+                        -- lead us out of here.
+                        Known 0 -> State.state (runState (continue 1)) >> interpret fetcher maxIter (k ())
+                        Known 1 -> State.state (runState (continue 0)) >> interpret fetcher maxIter (k ())
+                        n -> error ("I don't see how this could have happened: " <> show n)
+                    else do a <- State.state (runState (continue 0)) >> interpret fetcher maxIter (k ())
+                            put vm
+                            b <- State.state (runState (continue 1)) >> interpret fetcher maxIter (k ())
+                            return $ liftA2 (<>) a b
+               Nothing -> do a <- State.state (runState (continue 0)) >> interpret fetcher maxIter (k ())
+                             put vm
+                             b <- State.state (runState (continue 1)) >> interpret fetcher maxIter (k ())
+                             return $ liftA2 (<>) a b
+        Stepper.Wait q ->
+          do m <- liftIO (fetcher q)
+             State.state (runState m) >> interpret fetcher maxIter (k ())
+        Stepper.Note _ ->
+          -- simply ignore the note here
```

Comment:
Is the `Note` action actually required? It seems to be ignored in the interpreters in `UnitTest`, `SymExec` and `Fetch`?

Reply:
oh, I had missed `TTY.interpret`....